module/reports/helper.js

```diff
@@ -0,0 +1,557 @@
+                        })
+                        projectData.push(projectInfo);
+                    }));
+
+                    let data = {
+                        "reportType": returnTypeInfo[0].label,
+                        "projectDetails": projectData
+                    }
+                    let response = await dhitiService.viewFullReport(userToken, data);
+
+                    if (response && response.success == true) {
+
+                        return resolve({
+                            message: CONSTANTS.apiResponses.REPORT_GENERATED,
+                            result: {
+                                data: {
+                                    downloadUrl: response.data.pdfUrl
+                                }
+                            }
+                        });
+                    }
+
+                } else {
+                    await Promise.all(
+                        projectDetails.map(async projectList => {
+
+                            let reponseObj = {
+                                title: {
+                                    text: projectList.title
+                                },
+                                series: [{
+                                    name: projectList.title,
+                                    data: []
+                                }],
+                                xAxis: {
+
+                                }
+                            };
+                            reponseObj.series[0].data = [];
+                            reponseObj.xAxis.min = "";
+                            reponseObj.xAxis.max = "";
+                            reponseObj.series[0].name = projectList.title;
+                            if (projectList.tasks && projectList.tasks.length > 0) {
+                                await Promise.all(projectList.tasks.map(async taskList => {
+
+                                    let status = taskList.status;
+
+                                    if (reponseObj.xAxis.min != "" && reponseObj.xAxis.max != "") {
+                                        if (moment(reponseObj.xAxis.min) > moment(taskList.startDate)) {
+                                            reponseObj.xAxis.min = taskList.startDate;
+                                        }
+                                        if (moment(reponseObj.xAxis.max) > moment(taskList.endDate)) { } else {
+                                            reponseObj.xAxis.max = taskList.endDate;
+                                        }
+                                    } else {
+                                        reponseObj.xAxis.min = taskList.startDate;
+                                        reponseObj.xAxis.max = taskList.endDate;
+                                    }
+
+                                    let color = "";
+                                    if (status == CONSTANTS.common.NOT_STARTED_STATUS) {
+                                        color = "#f5f5f5";
```

Comment:
@rakeshSgr - Move all the color codes to constant with name which explains what is the code for?